src/main/java/org/openrewrite/java/spring/cloud2022/MigrateRequestMappingOnFeignClient.java

```diff
@@ -0,0 +1,149 @@
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.cloud2022;
+
+import java.util.List;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.search.FindAnnotations;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.service.AnnotationService;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JLeftPadded;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.java.tree.TypeUtils;
+import org.openrewrite.marker.Markers;
+
+import static java.util.Collections.emptyList;
+import static org.openrewrite.Tree.randomId;
+
+public class MigrateRequestMappingOnFeignClient extends Recipe {
+
+    private static final String FEIGN_CLIENT = "org.springframework.cloud.openfeign.FeignClient";
+
+    private static final String REQUEST_MAPPING = "org.springframework.web.bind.annotation.RequestMapping";
+
+    @Override
+    public String getDisplayName() {
+        return "Migrate `@RequestMapping` on `FeignClient` to `@FeignClient` path attribute";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Support for `@RequestMapping` over a `FeignClient` interface was removed in Spring Cloud OpenFeign 2.2.10.RELEASE.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(Preconditions.and(
+                new UsesType<>(FEIGN_CLIENT, false),
+                new UsesType<>(REQUEST_MAPPING, false)),
+            new JavaIsoVisitor<ExecutionContext>() {
+                @Override
+                public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+                    if (!FindAnnotations.find(classDecl, "@" + FEIGN_CLIENT).isEmpty() &&
+                        !FindAnnotations.find(classDecl, "@" + REQUEST_MAPPING).isEmpty()) {
+                        List<J.Annotation> annotations = service(AnnotationService.class).getAllAnnotations(getCursor());
+                        J.Annotation requestMapping = annotations.stream()
+                            .filter(a -> TypeUtils.isOfClassType(a.getType(), REQUEST_MAPPING))
+                            .findFirst().orElse(null);
+                        J.Annotation feignClient = annotations.stream()
+                            .filter(a -> TypeUtils.isOfClassType(a.getType(), FEIGN_CLIENT))
+                            .findFirst().orElse(null);
```

Comment:
`@RequestMapping` and `@FeignClient` are not repeatable Annotations, therefor its is additional work to use FindAnnotations and afterward filter the annotations by hand. 
I suggest once filter the annotations and then decide.
```java
classDecl.getLeadingAnnotations().stream()
  .filter(a -> TypeUtils.isOfClassType(a.getType(), REQUEST_MAPPING))
                            .findFirst().orElse(null)
//...
if (requestMapping != null && feignClient != null) {
}

```
This way it omits running a full blown Visitor for Informations already at hand ;)